pkgs/by-name/ml/mlx42/package.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, glfw
+, enableDebug ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mlx42";
+  version = "2.3.2";
+
+  src = fetchFromGitHub {
+    owner = "codam-coding-college";
+    repo = "MLX42";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-JCBV8NWibSugqXkbgP0v3gDfaaMNFYztWpBRfHJUG8E=";
+  };
+
+  postPatch = ''
+    patchShebangs ./tools
+    # Build shared library instead of static library
+    substituteInPlace CMakeLists.txt \
```

Comment:
```suggestion
    patchShebangs ./tools
  ''
  # Build shared library instead of static library
  + lib.optionalString enableShared ''
    substituteInPlace CMakeLists.txt \
```